### ISSUE TITLE
Unknown word CssSyntaxError in import inside script tag in vue file

### ISSUE THREAD

**Issue by eldair** [stylelint/stylelint#5634]
> Clearly describe the bug

CssSyntaxError unknown word in import inside script tag in vue file
![image](https://user-images.githubusercontent.com/470003/138448882-42a7f734-e6e0-43c8-8bca-5d25d7f3930e.png)

> Which rule, if any, is the bug related to?

> What code is needed to reproduce the bug?
```
<script>
    import {ref, nextTick, onBeforeUnmount, watch, inject} from 'vue';
</script>
```
> What Stylelint configuration is needed to reproduce the bug?

> Which version of Stylelint are you using?

e.g. `14.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

VSCode via stylelint plugin

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors did you get)?

CssSyntaxError unknown word in import inside script tag in vue file

**Comment by ashmenon** [comment]
I was struggling with this after upgrading, and @ybiquitous is correct. Also, if you need to use multiple syntaxes (e.g. if you have both SCSS and HTML-like files), you can use the `overrides` property as well.

In my `.stylelintrc.js` file I have the following:
```
customSyntax: "postcss-scss",
overrides: [
	{
	  "files": ["**/*.vue"],
	  "customSyntax": "postcss-html"
	},
],
```

From what I understand, [`postcss-syntax`](https://github.com/gucong3000/postcss-syntax) might be able to handle this, but I've already wasted enough time getting this to work so this is good enough for me lol.

**Comment by m-shum** [comment]
Still having this issue in our year of the lord 2022. 
Having spent almost three hours looking through the docs for every extension and plugin, I'm not really sure what to do. 

In vue3, I'm using vite-plugin-stylelint, which is included in vite.config.
My stylelint config is as follows:
```
{
  "customSyntax": "postcss-html",
  "extends": [
    "stylelint-config-html",
    "stylelint-config-recommended-scss",
    "stylelint-config-recommended-vue/scss",
    "stylelint-prettier/recommended"
  ],
  "overrides": [
    {
      "files": ["**/*.vue"],
      "customSyntax": "postcss-scss"
    }
  ],
  "plugins": ["stylelint-order", "stylelint-prettier"],
  "rules": {
    "declaration-empty-line-before": "never",
    "font-family-name-quotes": "always-unless-keyword",
    "order/order": ["custom-properties", "declarations"],
    "order/properties-alphabetical-order": true,
    "prettier/prettier": true,
    "selector-class-pattern": [
      "/^[a-z]+(?:-[a-z]+)*(?:_{2}[a-z]+(?:-[a-z]+)*)?(?:-{2}[a-z]+(?:-[a-z]+)*)?$/",
      {
        "resolveNestedSelectors": true
      }
    ]
  }
}
```

I've added 'vue' to stylelint.validate in the extension settings.

Unfortunately my components are still flagging vue syntax as unrecognised AND stylelint-order is picking up errors but not fixing them unless I run `npx stylelint --fix src/file/path`, which is something that's always been handled on save. 

If anyone can shed light on what I'm doing wrong, I will owe you my undying loyalty and my soul

**Comment by ybiquitous** [comment]
@m-shum Could you share your code for reproduction or provide a minimal reproduction repository? It might help us identify your problem.

**Comment by m-shum** [comment]
> @m-shum Could you share your code for reproduction or provide a minimal reproduction repository? It might help us identify your problem.

Here's the base repo: https://github.com/m-shum/test

**Comment by ybiquitous** [comment]
@m-shum Thanks. But I cannot see CssSyntaxError even when running `npm run lint:stylelint`. 🤔

**Comment by jeddy3** [comment]
> Here's the base repo: https://github.com/m-shum/test

I'm enjoying codespaces for running reproducible repos in the browser!

Like @ybiquitous, I get no `CssSyntaxError` when running on the CLI.

Can you share your VSCode config file, please?

You can also simplify your Stylelint config to:

```js
module.exports = {
  extends: [
    'stylelint-config-standard-scss',
    'stylelint-config-recommended-vue/scss',
    'stylelint-config-prettier-scss',
  ],
  plugins: ['stylelint-order'],
  rules: {
    'declaration-empty-line-before': 'never',
    'font-family-name-quotes': 'always-unless-keyword',
    'order/order': ['custom-properties', 'declarations'],
    'order/properties-alphabetical-order': true,
    'selector-class-pattern': [
      /^[a-z]+(?:-[a-z]+)*(?:_{2}[a-z]+(?:-[a-z]+)*)?(?:-{2}[a-z]+(?:-[a-z]+)*)?$/,
      {
        resolveNestedSelectors: true,
      },
    ],
    'value-keyword-case': [
      'lower',
      {
        ignoreKeywords: [/ease(?:[A-Z][a-z]+){2,}/],
      },
    ],
  },
}
```

1. the configs you're extending set the `customSyntax` property, so you don't have to.
2. it's [generally not recommended](https://prettier.io/docs/en/integrating-with-linters.html#notes) to run Prettier inside Stylelint

**Comment by m-shum** [comment]
@jeddy3 @ybiquitous The CssSyntaxError doesn't show in CLI – it will show in your editor: 
<img width="618" alt="image" src="https://user-images.githubusercontent.com/62724405/205081313-96c75099-34d0-41cc-aeaf-7993b487dd80.png">

I added a copy of my vscode config file to the repo.

@ybiquitous Which version of the vscode stylelint plugin are you using? I updated to the latest, updated my plugin configuration as advised in the migration guide, and it's still not fixing on save or linting vue files correctly. 

Relevant config snippets:

```
  "editor.formatOnSave": true,
  "stylelint.enable": true,
  "stylelint.config": {},
  "stylelint.snippet": ["css", "less", "postcss", "scss", "vue"],
  "stylelint.validate": ["css", "less", "postcss", "vue", "scss"],
  ```
  and
  
  ```
    "editor.codeActionsOnSave": {
    "source.fixAll": true,
    "source.organizeImports": true
  }
  ```

**Comment by jeddy3** [comment]
> "stylelint.config": {},

You'll need to remove this line. You're setting an empty Stylelint configuration object for VSCode. The [default is `null`, not `{}`](https://marketplace.visualstudio.com/items?itemName=stylelint.vscode-stylelint#stylelint.validate).

**Comment by jeddy3** [comment]
I'll open to issue to discuss whether we can add more validation to the configuration object.

**Comment by m-shum** [comment]
> > "stylelint.config": {},
> 
> You'll need to remove this line. You're setting an empty Stylelint configuration object for VSCode. The [default is `null`, not `{}`](https://marketplace.visualstudio.com/items?itemName=stylelint.vscode-stylelint#stylelint.validate).

Oh my god. You are a literal angel. Thank you thank you thank you!

**Comment by jeddy3** [comment]
> I'll open to issue to discuss whether we can add more validation to the configuration object.

We surface an error already for CLI, e.g.

```
// .stylelintrc
{}
```

Gives:

```console
Error: No rules found within configuration. Have you provided a "rules" property?
```

It seems the VS Code extension doesn't surface the error, though. One for that issue tracker, I think.

> Thank you thank you thank you!

No worries.

**Comment by jackchoumine** [comment]
> @jeddy3 @ybiquitous The CssSyntaxError doesn't show in CLI – it will show in your editor: <img alt="image" width="618" src="https://user-images.githubusercontent.com/62724405/205081313-96c75099-34d0-41cc-aeaf-7993b487dd80.png">
> 
> I added a copy of my vscode config file to the repo.
> 
> @ybiquitous Which version of the vscode stylelint plugin are you using? I updated to the latest, updated my plugin configuration as advised in the migration guide, and it's still not fixing on save or linting vue files correctly.
> 
> Relevant config snippets:
> 
> ```
>   "editor.formatOnSave": true,
>   "stylelint.enable": true,
>   "stylelint.config": {},
>   "stylelint.snippet": ["css", "less", "postcss", "scss", "vue"],
>   "stylelint.validate": ["css", "less", "postcss", "vue", "scss"],
> ```
> 
> and
> 
> ```
>   "editor.codeActionsOnSave": {
>   "source.fixAll": true,
>   "source.organizeImports": true
> }
> ```

hello bro, What theme do u use? it is so nice.

**Comment by jonathanandrewsuk** [comment]
> > "stylelint.config": {},
> 
> You'll need to remove this line. You're setting an empty Stylelint configuration object for VSCode. The [default is `null`, not `{}`](https://marketplace.visualstudio.com/items?itemName=stylelint.vscode-stylelint#stylelint.validate).

Thank you, this has been bugging me for days!

**Comment by BonjourYY** [comment]
> > "stylelint.config": {},
> 
> You'll need to remove this line. You're setting an empty Stylelint configuration object for VSCode. The [default is `null`, not `{}`](https://marketplace.visualstudio.com/items?itemName=stylelint.vscode-stylelint#stylelint.validate).

thank you, following your method, I have solved the same problem.I want to know why this issue occurred and if it is a bug. After installing the extensions, I didn't do anything,  Is the `"stylelint.config": {},` automatically added?

**Comment by grmculmfuer9** [comment]
I am having the same issue, could you help me solve it?

**Comment by harsharavindra** [comment]
I am facing the same issue as well.. Did anyone find the soultion ?

**Comment by heyakyra** [comment]
Commenting here to help anyone coming from search engines. I hit this in a stupid way for an unrelated plugin (`unplugin-vue`) which is that VSCode was inserting `// …` for CSS/SCSS/SASS comments in SFC files instead of `/* … */` which I only discovered due to adding more unwittingly-invalid "comments" and seeing the number of errors keep rising.
